src/Microsoft.NetCore.Analyzers/Core/ImmutableCollections/DoNotCallToImmutableCollectionOnAnImmutableCollectionValue.cs

```diff
@@ -39,30 +48,36 @@ public override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationStartAction(compilationStartContext =>
             {
-                var immutableArrayType = compilationStartContext.Compilation.GetTypeByMetadataName(ImmutableArrayMetadataName);
-                if (immutableArrayType == null)
+                var compilation = compilationStartContext.Compilation;
+                if (compilation.GetTypeByMetadataName(ImmutableArrayMetadataName) == null)
                 {
                     return;
                 }
 
                 compilationStartContext.RegisterOperationActionInternal(operationContext =>
                 {
                     var invocation = (IInvocationExpression)operationContext.Operation;
-                    if (invocation.TargetMethod?.Name != "ToImmutableArray")
+                    var targetMethod = invocation.TargetMethod;
+                    if (targetMethod == null || !ImmutableCollectionMetadataNames.TryGetValue(targetMethod.Name, out string metadataName))
                     {
                         return;
                     }
 
+                    Debug.Assert(!string.IsNullOrEmpty(metadataName));
+                    var immutableCollectionType = compilation.GetTypeByMetadataName(metadataName);
+
```

Comment:
Please either add a null check here for `immutableCollectionType`. The check above only verified if ImmutableArray is defined, but the user might be running against a custom system assembly that doesn't define other immutable types.

Reply:
@mavasani I don't need to add tests for this, right? (The tests automatically reference the S.C.I assembly, so when I add that branch it'll never be covered.)

Reply:
You can potentially do so by creating a compilation that doesn't reference System.Collections.Immutable.dll, and the test source defines a user defined type named `System.Collections.Immutable.ImmutableArray`